### PR TITLE
server: remove deviation influence and remodel balance region schedule

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?
issue #3613

### What is changed and how it works?
I think deviation is not usable in low disk space, so remove deviation influence .
I enlarge operator influence to speed down schedule so it will not balance around.
I will put  operator that finsh duration less than 10s to fast operator queue,so the available and region size will be same .
I will check influence to avaid A->B,B->C,C->A . 
finally,I module balance region by matlab, so it will explain some situation

### Check List

Tests

#### past
available
image

available(simulink)
image

schedule
image

schedule(simulink)
image

#### Now
available(cluster init)
image
schedule(cluster init)
image

available(write)
image

schedule(write)
image

available(write simulink)
image

available(new store up)
image

available(new store up simulink)

schedule(new store up)
image

- Integration test
- Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
enhance balance region stability to avoid disk used up earlier

```

### CODE REVIEW

File: server/core/store_stats.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type storeStats struct {
 func newStoreStats() *storeStats {
 	return &storeStats{
 		rawStats:                 &pdpb.StoreStats{},
-		avgAvailable:             movingaverage.NewHMA(240),       // take 40 minutes sample under 10s heartbeat rate
+		avgAvailable:             movingaverage.NewHMA(60),        // take 40 minutes sample under 10s heartbeat rate
```
Comment (nolouch) [inline server/core/store_stats.go:41]:
```suggestion
		avgAvailable:             movingaverage.NewHMA(60),        // take 10 minutes sample under 10s heartbeat rate
```

**Comment by bufferflies** [comment]
PTAL @disksing

File: server/schedule/filter/filters.go [hunk 2]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+func (f *ScoreFilter) Type() string {
+	return "score-filter"
+}
+
+// Source return true
+func (f *ScoreFilter) Source(opt *config.PersistOptions, _ *core.StoreInfo) bool {
+	return true
+}
+
+// Target return true if target score greater than src
```
Comment (disksing) [inline server/schedule/filter/filters.go:812]: I think it should be `target score is less than src`.

File: server/schedule/filter/filters.go [hunk 3]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+
+// NewScoreFilter creates a Filter that filters all high score stores.
+func NewScoreFilter(scope string, sources *core.StoreInfo, opt *config.PersistOptions) Filter {
+	return &ScoreFilter{
+		scope: scope,
+		score: sources.RegionScore(opt.GetRegionScoreFormulaVersion(), opt.GetHighSpaceRatio(), opt.GetLowSpaceRatio(), 0, 0),
+	}
+}
+
+// Scope scope
```
Comment (disksing) [inline server/schedule/filter/filters.go:797]: incomplete comment.

File: server/schedule/filter/filters.go [hunk 4]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
 	}, copyLeader, opts...)
 	return cloneRegion
 }
+
+// ScoreFilter filter store that score higher than src
+type ScoreFilter struct {
```
Comment (disksing) [inline server/schedule/filter/filters.go:784]: RegionScoreFilter should be more suitable

File: server/schedule/filter/filters.go [hunk 5]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+	scope string
+	score float64
+}
+
+// NewScoreFilter creates a Filter that filters all high score stores.
+func NewScoreFilter(scope string, sources *core.StoreInfo, opt *config.PersistOptions) Filter {
```
Comment (disksing) [inline server/schedule/filter/filters.go:790]: sources -> source

File: server/schedule/filter/filters.go [hunk 6]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+		score: sources.RegionScore(opt.GetRegionScoreFormulaVersion(), opt.GetHighSpaceRatio(), opt.GetLowSpaceRatio(), 0, 0),
+	}
+}
+
+// Scope scope
+func (f *ScoreFilter) Scope() string {
+	return f.scope
+}
+
+// Type type
```
Comment (disksing) [inline server/schedule/filter/filters.go:802]: incomplete comment.

File: server/schedule/filter/filters.go [hunk 7]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+func (f *ScoreFilter) Type() string {
+	return "score-filter"
+}
+
+// Source return true
```
Comment (disksing) [inline server/schedule/filter/filters.go:807]: incomplete comment.

File: server/schedule/filter/filters.go [hunk 8]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
 	}, copyLeader, opts...)
 	return cloneRegion
 }
+
+// ScoreFilter filter store that score higher than src
```
Comment (disksing) [inline server/schedule/filter/filters.go:783]: Better to mention that it only filters target store.

File: server/schedule/operator_controller.go [hunk 9]
```diff
@@ -52,6 +52,8 @@ var (
 	PushOperatorTickInterval = 500 * time.Millisecond
 	// StoreBalanceBaseTime represents the base time of balance rate.
 	StoreBalanceBaseTime float64 = 60
+	// FastOperatorFinishTime min finish time,if finish duration less than it,op will be pushed to fast operator queue
```
Comment (disksing) [inline server/schedule/operator_controller.go:55]: add space after `,`.

File: server/core/store.go [hunk 10]
```diff
@@ -305,6 +305,16 @@ func (s *StoreInfo) regionScoreV2(delta int64, deviation int, lowSpaceRatio floa
 	A := float64(float64(s.GetAvgAvailable())-float64(deviation)*float64(s.GetAvailableDeviation())) / gb
 	C := float64(s.GetCapacity()) / gb
 	R := float64(s.GetRegionSize() + delta)
+	if R < 0 {
```
Comment (disksing) [inline server/core/store.go:308]: em, why is `deviation` not removed? (Line 305)

Comment (bufferflies) [reply server/core/store.go:308]: I think the stragy can config before, so it could roll back after it online

File: server/schedulers/utils.go [hunk 11]
```diff
@@ -84,17 +85,32 @@ func (p *balancePlan) shouldBalance(scheduleName string) bool {
...
+	// to avoid schedule too much, if A's score less than B and C in small range,
+	// we want that A can be moved in one region not two
 	targetInfluence := p.GetOpInfluence(targetID)
-	sourceDelta, targetDelta := sourceInfluence-tolerantResource, targetInfluence+tolerantResource
+	// to avoid schedule call back
+	// A->B, A's influence is negative，so A will be target,C may move region to A
+	if targetInfluence < 0 {
+		targetInfluence = -targetInfluence
+	}
+	//sourceDelta, targetDelta := sourceInfluence-tolerantResource, targetInfluence+tolerantResource
```
Comment (disksing) [inline server/schedulers/utils.go:103]: Should remove this line. The rest LGTM.

**Review by disksing — APPROVED** [review]

File: server/schedule/filter/filters.go [hunk 12]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+
+// NewRegionScoreFilter creates a Filter that filters all high score stores.
+func NewRegionScoreFilter(scope string, source *core.StoreInfo, opt *config.PersistOptions) Filter {
+	return &RegionScoreFilter{
+		scope: scope,
+		score: source.RegionScore(opt.GetRegionScoreFormulaVersion(), opt.GetHighSpaceRatio(), opt.GetLowSpaceRatio(), 0),
+	}
+}
+
+// Scope scope only for balance region
```
Comment (lhy1024) [inline server/schedule/filter/filters.go:797]:
```suggestion
// Scope scopes only for balance region
```

File: server/schedule/filter/filters.go [hunk 13]
```diff
@@ -779,3 +779,38 @@ func createRegionForRuleFit(startKey, endKey []byte,
...
+func (f *RegionScoreFilter) Scope() string {
+	return f.scope
+}
+
+// Type type region score filter
```
Comment (lhy1024) [inline server/schedule/filter/filters.go:802]: ditto

**Review by nolouch — APPROVED** [review]
lgtm!

**Pull request merged**